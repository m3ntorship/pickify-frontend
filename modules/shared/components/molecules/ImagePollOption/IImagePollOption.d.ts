import type { IVoteIcon } from '../../atoms/VoteIcon/IVoteIcon';

declare namespace IImagePollOption {
  export interface IProps extends IVoteIcon.IProps {
    imageUrl: string;
    imgCaption: string;
    imgCaptionLetter?: string;
  }
}
export { IImagePollOption };
