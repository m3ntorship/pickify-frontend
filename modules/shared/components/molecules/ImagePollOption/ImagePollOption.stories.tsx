import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import ImagePollOption from './ImagePollOption';
import type { IImagePollOption } from './IImagePollOption';

export default {
  component: ImagePollOption,
  title: 'Molecules/IImagePollOption',
  argTypes: {
    percentage: {
      control: {
        type: 'range',
        min: 0,
        max: 100,
        step: 1,
      },
    },
  },
};
const Template: Story<IImagePollOption.IProps> = (args): ReactElement => (
  <div className="flex">
    <ImagePollOption {...args} />
  </div>
);
export const Default = Template.bind({});
Default.args = {
  media: [{ url: 'https://placeimg.com/640/480/any' }],
  imgCaption: 'caption 1',
  imgCaptionLetter: 'A',
  optionId: 'option_1',
  leastVoted: false,
  mostVoted: false,
  percentage: 80,
  isVoted: true,
};
