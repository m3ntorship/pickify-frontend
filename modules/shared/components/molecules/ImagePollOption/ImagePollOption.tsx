import React from 'react';
import type { FC, ReactElement } from 'react';
import Image from 'next/image';
import type { IImagePollOption } from './IImagePollOption';
import ImagePollCovered from '../../atoms/ImagePollCovered/ImagePollCovered';
import ImageCaption from '../ImageCaption/ImageCaption';
import ImagePollUncovered from '../ImagePollUncovered/ImagePollUncovered';
import { apiUrls } from '../../../configuration/ConfigPostCreation/config';

const ImagePollOption: FC<IImagePollOption.IProps> = ({
  media,
  imgCaption,
  imgCaptionLetter,
  optionId,
  leastVoted,
  mostVoted,
  percentage,
  isVoted,
  onOptionClick,
}): ReactElement => {
  return (
    <div className="relative w-auto" data-testid="image-poll-option">
      {media.length !== 0 ? (
        <>
          {media.map((image) => (
            <div key={image.url} className="relative">
              <div className="absolute w-full h-full rounded-md overflow-hidden">
                <Image
                  key={image.url}
                  src={`${apiUrls.mediaAPI}${image.url}`}
                  layout="responsive"
                  className="filter blur-sm"
                  objectFit="cover"
                  width={600}
                  height={600}
                />
              </div>
              <Image
                src={`${apiUrls.mediaAPI}${image.url}`}
                layout="responsive"
                className="rounded-md"
                objectFit="contain"
                width={600}
                height={600}
              />
            </div>
          ))}
          <div className="absolute bottom-4 right-4">
            {isVoted ? (
              <ImagePollUncovered
                id={optionId}
                type="vertical"
                leastVoted={leastVoted}
                mostVoted={mostVoted}
                percentage={percentage}
                verticalMeterHeight={150}
              />
            ) : (
              <ImagePollCovered
                isOneImageVote={false}
                onOptionClick={onOptionClick}
                id={optionId}
              />
            )}
          </div>
          <div className="absolute bottom-4 left-4">
            <ImageCaption
              imgCaption={imgCaption}
              imgCaptionLetter={imgCaptionLetter}
            />
          </div>
        </>
      ) : (
        <p>cannot display this image :(</p>
      )}
    </div>
  );
};

export default ImagePollOption;
