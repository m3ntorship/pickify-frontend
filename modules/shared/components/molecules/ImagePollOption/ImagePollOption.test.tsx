import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ImagePollOption from './ImagePollOption';

describe('ImagePollOption', () => {
  it('should render heart icon when passing isVoted=[false]', () => {
    const tree = renderer
      .create(
        <ImagePollOption
          imageUrl="https://source.unsplash.com/random"
          imgCaption="caption"
          imgCaptionLetter="A"
          optionId="option_1"
          leastVoted={false}
          mostVoted={false}
          percentage={0}
          isVoted={false}
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render primary progress bar when passing isVoted=[true] and mostVoted=[true]', () => {
    const tree = renderer
      .create(
        <ImagePollOption
          imageUrl="https://source.unsplash.com/random"
          imgCaption="caption"
          imgCaptionLetter="A"
          optionId="option_1"
          leastVoted={false}
          mostVoted
          percentage={80}
          isVoted
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render red progress bar when passing isVoted=[true] and leastVoted=[true]', () => {
    const tree = renderer
      .create(
        <ImagePollOption
          imageUrl="https://source.unsplash.com/random"
          imgCaption="caption"
          imgCaptionLetter="A"
          optionId="option_1"
          leastVoted
          mostVoted={false}
          percentage={20}
          isVoted
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
