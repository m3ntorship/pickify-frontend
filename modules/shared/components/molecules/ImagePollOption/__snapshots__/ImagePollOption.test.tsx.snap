// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePollOption should render heart icon when passing isVoted=[false] 1`] = `
<div
  className="relative w-auto"
  data-testid="image-poll-option"
>
  <div
    className="image-view-wrapper"
  >
    <div
      className="bg-image-box"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="bg-image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="option"
        className="image"
        decoding="async"
        id="https://placeimg.com/640/480/any"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="absolute bottom-4 right-4"
  >
    <div
      className="flex justify-end"
    >
      <div
        className="group"
      >
        <button
          className="btn btn-like btn-medium btn-rounded"
          disabled={false}
          id="option_1"
          onClick={[Function]}
          type="button"
        >
          <span>
            <svg
              className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
              data-file-name="SvgFilledHeart"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <div
      className="captionContent"
    >
      <span
        className=""
      >
        A
      </span>
      <div
        className="captionDivider "
      />
      <span
        className="image-caption"
        dir="auto"
        title="caption"
      >
        caption
      </span>
    </div>
  </div>
</div>
`;

exports[`ImagePollOption should render primary progress bar when passing isVoted=[true] and mostVoted=[true] 1`] = `
<div
  className="relative w-auto"
  data-testid="image-poll-option"
>
  <div
    className="image-view-wrapper"
  >
    <div
      className="bg-image-box"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="bg-image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="option"
        className="image"
        decoding="async"
        id="https://placeimg.com/640/480/any"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="absolute bottom-4 right-4"
  >
    <div
      className="vertical-vote-meter-wrapper"
      id="option_1"
    >
      <div
        className="percentage-wrapper"
      >
        <p
          className="percentage"
        >
          ðŸŒŸ
          <span>
            80
            %
          </span>
        </p>
      </div>
      <div
        className="flex h-full"
      >
        <div
          className="outer-vertical wrapper"
          style={
            Object {
              "height": "214.4px",
            }
          }
        >
          <div
            className="inner-vertical primary"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <div
      className="captionContent"
    >
      <span
        className=""
      >
        A
      </span>
      <div
        className="captionDivider "
      />
      <span
        className="image-caption"
        dir="auto"
        title="caption"
      >
        caption
      </span>
    </div>
  </div>
</div>
`;

exports[`ImagePollOption should render red progress bar when passing isVoted=[true] and leastVoted=[true] 1`] = `
<div
  className="relative w-auto"
  data-testid="image-poll-option"
>
  <div
    className="image-view-wrapper"
  >
    <div
      className="bg-image-box"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="option"
          className="bg-image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="option"
        className="image"
        decoding="async"
        id="https://placeimg.com/640/480/any"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="absolute bottom-4 right-4"
  >
    <div
      className="vertical-vote-meter-wrapper"
      id="option_1"
    >
      <div
        className="percentage-wrapper"
      >
        <p
          className="percentage"
        >
          <span>
            20
            %
          </span>
        </p>
      </div>
      <div
        className="flex h-full"
      >
        <div
          className="outer-vertical wrapper"
          style={
            Object {
              "height": "53.6px",
            }
          }
        >
          <div
            className="inner-vertical error"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <div
      className="captionContent"
    >
      <span
        className=""
      >
        A
      </span>
      <div
        className="captionDivider "
      />
      <span
        className="image-caption"
        dir="auto"
        title="caption"
      >
        caption
      </span>
    </div>
  </div>
</div>
`;
